File: Planification/src/models/CompetitionDAO.java
package models;

import entities.Competition;
import entities.Film;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;

public class CompetitionDAO extends DAO
{
    public List<Competition> getCompetitions() // on recupère toutes les compétitions
    {
        List<Competition> competitions = new ArrayList<Competition>();
        
        ResultSet result = super.query("SELECT id, libelle FROM competition");
        
        try
        {
            while(result.next())
            {
                competitions.add(new Competition(result.getInt("id"), result.getString("libelle")));
            }
        }
        catch (SQLException e)
        {
            System.err.println("Error : " + e.getMessage());
        }
        
        return competitions;
    }
    
    public Competition getCompetition(int idCompetition) // On récupère une compétition
    {
        List<String> args = new ArrayList<String>();
        args.add("" + idCompetition);
        ResultSet result = super.query("SELECT id, libelle FROM competition WHERE id = ?", args);
        
        try 
        {
            result.first();
            return new Competition(result.getInt("id"), result.getString("libelle"));
        } 
        catch (SQLException e)
        {
            System.err.println("Error : " + e.getMessage());
        }
        
        return null;
    }
            
}

File: Hebergement/models/CompetitionDAO.php
<?php

require_once(PATH_MODELS.'DAO.php');
require_once(PATH_ENTITY.'Competition.php');

class CompetitionDAO extends DAO
{
  public function getCompetitions()
  {
    $res = $this->queryAll('SELECT * FROM Competition');

    if($res)
    {
      foreach($res as $line)
      {
        $competitions[] = new Competition($line['id'], $line['libelle']);
      }
      return $competitions;
    }
    else return array();
  }

  public function getCompetition($id)
  {
    $res = $this->queryRow('SELECT * FROM Competition WHERE id = ?', array($id));

    if($res)
    {
      return new Competition($res['id'], $res['libelle']);
    }
    else return null;
  }
}

 ?>
